pkg/operator/staticpod/controller/installer/installer_controller.go

```diff
@@ -384,6 +393,42 @@ func nodeToStartRevisionWith(ctx context.Context, getStaticPodStateFn staticPodS
 	return 0, reason, nil
 }
 
+// nodeStatusesFilteredForRevision returns a list of nodeStatuses which have been selected for the next revision.
+func nodeStatusesFilteredForRevision(ctx context.Context, nodeFilterFn func(ctx context.Context) (map[string]bool, error), nodeStatuses []operatorv1.NodeStatus) ([]operatorv1.NodeStatus, error) {
+	if nodeFilterFn == nil {
+		return nodeStatuses, nil
+	}
+
+	nodeSelectorHostnameMap, err := nodeFilterFn(ctx)
+	if err != nil {
+		return nil, err
+	}
+	if len(nodeSelectorHostnameMap) == 0 {
+		return nodeStatuses, nil
```

Comment:
I can't exactly think of when the [etcd-operator's filter](https://github.com/openshift/cluster-etcd-operator/blob/7951acdaf4a8ebcffc5cfbf31e5f5bd184b5285d/pkg/operator/staticpodnodefilter/staticpodnodefilter.go#L13) would return an empty node map (maybe some node list error not caught by the filter), but if for whatever reason it does, do we want to resort to the default of no-filter, or error out saying the filter is invalid?
```suggestion
	if len(nodeSelectorHostnameMap) == 0 {
		return nil, fmt.Errorf("invalid node filter: the node map cannot be empty")
```

Reply:
And we are error-ing out below as well when the node map is not fully populated too. So maybe just the same error.

Reply:
good catch!